packages/test/test-end-to-end-tests/src/test/gc/gcStateResetInSummaries.spec.ts

```diff
@@ -0,0 +1,276 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import { ContainerRuntimeFactoryWithDefaultDataStore, DataObjectFactory } from "@fluidframework/aqueduct";
+import { TelemetryNullLogger } from "@fluidframework/common-utils";
+import { IContainer } from "@fluidframework/container-definitions";
+import {
+    ContainerRuntime,
+    IAckedSummary,
+    IContainerRuntimeOptions,
+    SummaryCollection,
+} from "@fluidframework/container-runtime";
+import { ISummaryContext } from "@fluidframework/driver-definitions";
+import {
+    ISummaryTree,
+    SummaryType,
+} from "@fluidframework/protocol-definitions";
+import { channelsTreeName, IGarbageCollectionSummaryDetails } from "@fluidframework/runtime-definitions";
+import { requestFluidObject } from "@fluidframework/runtime-utils";
+import { ITestObjectProvider } from "@fluidframework/test-utils";
+import { describeFullCompat } from "@fluidframework/test-version-utils";
+import { wrapDocumentServiceFactory } from "./gcDriverWrappers";
+import { loadSummarizer, TestDataObject, submitAndAckSummary } from "./mockSummarizerClient";
+
+describeFullCompat("GC state reset in summaries", (getTestObjectProvider) => {
+    let provider: ITestObjectProvider;
+    const dataObjectFactory = new DataObjectFactory(
+        "TestDataObject",
+        TestDataObject,
+        [],
+        []);
+
+    const defaultRuntimeOptions: IContainerRuntimeOptions = {
+        summaryOptions: { generateSummaries: false },
+    };
+
+    const logger = new TelemetryNullLogger();
+
+    // Stores the latest summary uploaded to the server.
+    let latestUploadedSummary: ISummaryTree | undefined;
+    // Stores the latest summary context uploaded to the server.
+    let latestSummaryContext: ISummaryContext | undefined;
+    // Stores the latest acked summary for the document.
+    let latestAckedSummary: IAckedSummary | undefined;
+
+    let mainContainer: IContainer;
+
+    /** Creates a new container with the GC enabled / disabled as per gcAllowed param. */
+    const createContainer = async (gcAllowed?: boolean): Promise<IContainer> => {
```

Comment:
This method seems like it might be nice to incorporate into the engine code. Do you think that's true?

Reply:
Not sure what you mean by engine code. Do you mean the TestObjectProvider?
Most of the tests create a container by calling provider.createContainer and passing in a runtime factory. This is a specific scenario where we want to toggle the gcAllowed flag and so the runtime factory is created inside this function.